server/routes/public.js

```diff
@@ -29,15 +29,48 @@ router.get('/api/login', async (req, res) => {
 });
 
 router.get('/api/create-user', async (req, res) => {
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
+       
+       //The following joi verifications make sure the values inputted adhere to the format requirements
+       //specified in the migrations> create-worker-table and migrations> create-student-table.
+
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        firstName: Joi.string().min(1).max(20).required(),
+        lastName: Joi.string().min(1).max(20).required(),
+        studentID: Joi.number().positive().integer(),
+        workerID: Joi.number().positive().integer(),
+        type: Joi.string().valid(['student', 'worker']),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+        accessToken: Joi.string().alphanum().min(3).max(30).required(), 
+        //insert joi verification for firstname,lastname, type,studentID,workerID,phone
+    });
+    // the endpoint `/api/create-user` accepts ALL parameters for both user types, 
+    //as listed in the migrations> create-worker-table and migrations> create-student-table
+
     const firstName = req.query.firstName;
     const lastName = req.query.lastName;
     const type = req.query.type; //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
-    const studentID = req.query.studentID;
+    const studentID = req.query.studentID;//Students only
+    const workerID = req.query.workerID;//Workers only
     const email = req.query.email;
     const password = req.query.password;
     const phone = req.query.phone;
+    const school_id = req.query.school_id; //Students only
+    const specialization = req.query.specialization; //Workers only
+    const accessToken = req.query.accessToken; //is this needed? If yes, add to worker.js line 18 and student.js line 17
+
+   
+
+    //error check
+    const { error, value } = paramSchema.validate({ email: paramEmail, password: paramPassword });
+    if (!_.isNil(error)) res.send(error);
+
 
-    const user = await accountHandler.createUserAccount(firstName, lastName, type, studentID, email, password, phone);
+    const user = await accountHandler.createUserAccount(firstName, lastName, type, studentID, email, password, phone,workerID,school_id,specialization,accessToken);
```

Comment:
Hi Nancy,
Again, make sure to keep variable naming consistent, e.g. `school_id` should be `schoolId`.

Reply:
Hi Amy,
Good catch!
My action: I've changed the parameter names inside the `accountHandler.createUserAccount` so that they are in sync with `005.do.create-worker-table.sql` and `004.do.add-student-table-data.sql.`

You can now see these changes in line 73 of server/routes/public.js

Please confirm if this takes care of your suggestion!

Reply:
Hi Nancy,
That's perfect. Thank you for making those changes!